pkg/io/jsonio/unmarshal.go

```diff
@@ -122,13 +125,19 @@ func parseCidrSegments(jsonSegments spec.SpecSegments, configDefs *ir.ConfigDefs
 			if err != nil {
 				return nil, err
 			}
-			segmentMap[c] = ir.CIDRDetails{
-				OverlappingVPCs:  vpcs,
-				ContainedSubnets: subnets,
+
+			cidrs = cidrs.Union(c)
+			for _, vpc := range vpcs {
+				overlappingVPCs[vpc] = struct{}{}
+			}
+			for _, subnet := range subnets {
+				containedSubnets[subnet] = struct{}{}
```

Comment:
Seems simpler to me to avoid using maps as sets here. Instead, concatenate vpcs and subnets into the slices `OverlappingVPCs` and `ContainedSubnets`, then outside of the loop run `slices.Compact(slices.Sorted(slices.Values(OverlappingVPCs)))`

Reply:
fixed